modules/core/CMakeLists.txt

```diff
@@ -107,6 +107,9 @@ endif()
 if(OPENCV_LIBVA_LINK)
   ocv_append_source_file_compile_definitions("${CMAKE_CURRENT_LIST_DIR}/src/va_intel.cpp" "OPENCV_LIBVA_LINK=1")
 endif()
+if(OPENCV_ENABLE_EGL_INTEROP)
+  ocv_append_source_file_compile_definitions("${CMAKE_CURRENT_LIST_DIR}/src/opengl.cpp" "HAVE_EGL_INTEROP")
```

Comment:
Perhaps `HAVE_EGL` and `HAVE_OPENGL` should be checked too.

Reply:
It does in the main [CMakeLists.txt](https://github.com/opencv/opencv/pull/22704/files/14d9a17c1a5a09efad8bea2c8cb4c8b55552afe1#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR326)

Reply:
Not really.

User could specify `WITH_OPENGL=ON` but there could be no OpenCL dev packages on build system (or for build target platform in case of cross-compiling).

`WITH_` / ` `OPENCV_ENABLE_` is a user intention (lets enable this if available).
Next, `find_package()` should be executed.
Preferable `try_compile()` should be run to validate dependency configuration.
After checks above `HAVE_` variable is set to `ON` / `1`.
Feature usage should be guarded by `HAVE_` checks.

All dependencies should be properly detected on configuration (cmake) stage.
Build stage (make) should not fail due to dependency misusing.

Reply:
I see now. Will probably create a PR today.